src/main/java/org/cryptomator/ui/mainwindow/ResizeController.java

```diff
@@ -40,61 +40,56 @@ public class ResizeController implements FxController {
-				//If the position is illegal, then the window appears on the main screen in the middle of the window.
-				Rectangle2D primaryScreenBounds = Screen.getPrimary().getBounds();
-				window.setX((primaryScreenBounds.getWidth() - window.getMinWidth()) / 2);
-				window.setY((primaryScreenBounds.getHeight() - window.getMinHeight()) / 2);
-				window.setWidth(window.getMinWidth());
-				window.setHeight(window.getMinHeight());
-			} else {
-				window.setHeight(settings.windowHeight.get() > window.getMinHeight() ? settings.windowHeight.get() : window.getMinHeight());
-				window.setWidth(settings.windowWidth.get() > window.getMinWidth() ? settings.windowWidth.get() : window.getMinWidth());
-				window.setX(settings.windowXPosition.get());
-				window.setY(settings.windowYPosition.get());
-			}
+		if (!neverTouched()) {
+			window.setHeight(settings.windowHeight.get() > window.getMinHeight() ? settings.windowHeight.get() : window.getMinHeight());
+			window.setWidth(settings.windowWidth.get() > window.getMinWidth() ? settings.windowWidth.get() : window.getMinWidth());
+			window.setX(settings.windowXPosition.get());
+			window.setY(settings.windowYPosition.get());
 		}
-		savePositionalSettings();
+		window.setOnShowing(this::checkDisplayBounds);
 	}
 
 	private boolean neverTouched() {
 		return (settings.windowHeight.get() == 0) && (settings.windowWidth.get() == 0) && (settings.windowXPosition.get() == 0) && (settings.windowYPosition.get() == 0);
 	}
 
-	private boolean didDisplayConfigurationChange() {
-		String currentDisplayConfiguration = getMonitorSizes();
-		String settingsDisplayConfiguration = settings.displayConfiguration.get();
-		boolean configurationHasChanged = !settingsDisplayConfiguration.equals(currentDisplayConfiguration);
-		if (configurationHasChanged) settings.displayConfiguration.set(currentDisplayConfiguration);
-		return configurationHasChanged;
-	}
-
-	private String getMonitorSizes() {
-		ObservableList<Screen> screens = Screen.getScreens();
-		StringBuilder sb = new StringBuilder();
-		for (int i = 0; i < screens.size(); i++) {
-			Rectangle2D screenBounds = screens.get(i).getBounds();
-			if (!sb.isEmpty()) sb.append(" ");
-			sb.append("displayId: " + i + ", " + screenBounds.getWidth() + "x" + screenBounds.getHeight() + ";");
+	private void checkDisplayBounds(WindowEvent evt) {
+		// Rect of half the height of the title bar
+		Rectangle2D titleBarRect = new Rectangle2D(window.getX(), window.getY(), window.getWidth(), titleBar.getMinHeight() / 2);
+
+		// Fraction f defines the minimum amount of the width/height that must be visible to not reset the window. Must be less or equal to 0.5
+		double f = 0.5;
+		double minX = window.getX() + f * window.getWidth();
+		double minY = window.getY() + f * window.getHeight();
+		double width = window.getWidth() - 2 * f * window.getWidth();
+		double height = window.getHeight() - 2 * f * window.getHeight();
+		Rectangle2D centerRect = new Rectangle2D(minX, minY, width, height);
```

Comment:
Don't use fractions here, but a fixed pixel threshold. It should not matter, how big the current window is. Only, if i can see at least XX pixels of of the window.

For the threshold, i suggest to use 10px.

Reply:
I changed it to 10px but I find it rather small. See the screenshot below. With 10px threshold this Cryptomator window will not be reset but it is barely visible. There is only a small area where you can select the title bar and drag it to a more useable place on the screen. I think this is not very user friendly and the value should be a lot bigger. Could also be different values for width and height.
![image](https://github.com/cryptomator/cryptomator/assets/14298143/87a23b39-a21d-4a08-b5e6-1d88aee23cb3)